From patchwork Sun May 11 20:12:35 2025
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Christian Marangi <ansuelsmth@gmail.com>
X-Patchwork-Id: 14084290
Return-Path: 
 <linux-mediatek-bounces+linux-mediatek=archiver.kernel.org@lists.infradead.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by smtp.lore.kernel.org (Postfix) with ESMTPS id 9BCA8C3ABC3
	for <linux-mediatek@archiver.kernel.org>;
 Sun, 11 May 2025 20:34:00 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20210309; h=Sender:List-Subscribe:List-Help
	:List-Post:List-Archive:List-Unsubscribe:List-Id:Content-Transfer-Encoding:
	MIME-Version:References:In-Reply-To:Message-ID:Date:Subject:To:From:Reply-To:
	Cc:Content-Type:Content-ID:Content-Description:Resent-Date:Resent-From:
	Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:List-Owner;
	bh=Cwoa6YEoLumzQE8cS+ad1dIlqWJKYevB3c6NtJXNM8U=; b=4hM2DZkbsJ1CdAo687CHlH6cFD
	h+nyJCydN9FFVVEoAZTW1KZ4d3OUcxe9Y85euRW6d4VhvDn+gEw4VSF7GsUV1v3RM3nh0o0B1/Z58
	fvMIKG+RcLn4+8tbwInnlCPCZaI+o8kaSRPoQpFmKtHMadbyxeUrHru3gev99EkRnYfuF6wIeJKQa
	06GrTJJK/OjM4wes7uTz7tP0SOMgKZmSPn8H71TBfFHI71QxSgg4+fqgJN6uXgxYAsiLch4tlil/q
	5BM8K1LLubWPYu3NIRMnPganvlqwERGfykhKDJ6rhtN1lyP5oV7ib07lCNs+QDUMwesHEwDt1KY34
	2+PY4N/g==;
Received: from localhost ([::1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.98.2 #2 (Red Hat Linux))
	id 1uEDN1-00000007opZ-33NX;
	Sun, 11 May 2025 20:33:59 +0000
Received: from mail-wm1-x32a.google.com ([2a00:1450:4864:20::32a])
	by bombadil.infradead.org with esmtps (Exim 4.98.2 #2 (Red Hat Linux))
	id 1uED3I-00000007mre-3CA9;
	Sun, 11 May 2025 20:13:38 +0000
Received: by mail-wm1-x32a.google.com with SMTP id
 5b1f17b1804b1-43d0618746bso28318765e9.2;
        Sun, 11 May 2025 13:13:36 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20230601; t=1746994415; x=1747599215;
 darn=lists.infradead.org;
        h=content-transfer-encoding:mime-version:references:in-reply-to
         :message-id:date:subject:to:from:from:to:cc:subject:date:message-id
         :reply-to;
        bh=Cwoa6YEoLumzQE8cS+ad1dIlqWJKYevB3c6NtJXNM8U=;
        b=HRjXQFrJanUnATn2U7lbS4FN1+wvoIsuPpQhGnv39IIwT1BZd/FkFrwcHcvTf1kx/I
         ZZufhYesVl82cGiRDtzJNi424cvecnRN5W0ip3fFOMGEleR1VUSS/jyiGWkzd2IjPTAg
         lB78iNQc9ih+eE5stcJd5GpFGtRQc91dGKIXsW0abNqdMD8EuXQ/QqU+s7i8kEaCpawo
         iI1NKg0hSYcnDsR4xaajnoW09livHFr73rJdko74e5buK0ZiuGKAL1ygTyqrEC0c6/iA
         hqZG7fjswvHWrM1jCqyYLA/DJ0wWHl7ZaFpNFXjrova+j7GFXSNu70GiqS9M2VXi6zoj
         PUeQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20230601; t=1746994415; x=1747599215;
        h=content-transfer-encoding:mime-version:references:in-reply-to
         :message-id:date:subject:to:from:x-gm-message-state:from:to:cc
         :subject:date:message-id:reply-to;
        bh=Cwoa6YEoLumzQE8cS+ad1dIlqWJKYevB3c6NtJXNM8U=;
        b=Kg15+QvfY3Beo8LZpr1XBe6Qse6pr+dr7bPp6plN/pgYozCkur23E92jkVbNHnRw2s
         JNZDDpshB6PRivwN5mwoycV573OI2FqHWmUyOfKVAb0CuA04rMkTu+qoKMkBc+BFQUBn
         mUaOM+IldgtF6+SGjBz3XugynZ29s/PbjDT+4YQWeQmOMCyLQdjrw+vTddGjpfugdVM2
         Rhst3hW1DSxoY+Kg7RniEvE5v9NBfl+8navsnDLi54IKhqOp1lQKntB2PkZSlAHR81C1
         r5qRFlHAlFFDE6Df2bZVJa/FQxE2U9LDzYvL1f33BdIMDPJtlXSTA9JTpdNJCpY1xL1u
         mHJw==
X-Forwarded-Encrypted: i=1;
 AJvYcCWY7YXnhYQCbkCz1BfV13vnw9wtYaGaSr8jMhvIq9Ktsy/9J8DtkIfyNzxxYXIMaHjjT8LT1TJlXUeE8ozs+ew=@lists.infradead.org,
 AJvYcCXS4f6JxLChRHUCyIbD3XPqswA8z5RUeJV5LZo5EdAFMP40GrMhmqoqtutq/wgXzVH3zKBitfkJVZ1Pe7nRKqPW@lists.infradead.org
X-Gm-Message-State: AOJu0YwZfpT0MuiBdE3HWhsnVnPeiuZa/LDVZJ6L7kpRziGVoRQeJ3DI
	FYO6ZyQZ/Xnsp1+zmk52gdL8ktRABKJNfI+9Y7C40ZUpAzt5aakU
X-Gm-Gg: ASbGnctJMETyGV3i4aWy7aPHV6TUhGTS6yVos9iHx7ImtCoe6r6PQX7P3BoU1DPMmsy
	+wQnSxpRS6Xb+e6xq0ldOWL0Amdb58/pxmxKmOpALf3mPkos4B+rbhvjA8fqAghjlSCV8jogZPo
	ExW//52MmNLn7noWA+YB/fBlIyjx4K7vF46mHiVW+Q+JIoMJgze4Jr/1S/nKURt0bXJpM+u+VHT
	2tz19R01PO05bIkgYGL806UucDJtWjjUV0KM6U9rcTMk3QUE3hSDJz6Hufbu5kaPc0lXzlRi1/C
	6bw+UgBB0Nze3cf0vRrNYDwRPAReDKCVzzIkpVGPL3CPWPKqYa1cLRJpU9Ce79JjHUt36/btccm
	KZ+0D2nqy3T0APwrk14R0
X-Google-Smtp-Source: 
 AGHT+IEOWK8ud3qNK29LCO+35MG4PEuwn5Jd54n78utiERk537v3iwnnobOuImhQnqz6zBJyk8U4wg==
X-Received: by 2002:a05:600c:8205:b0:43c:eea9:f45d with SMTP id
 5b1f17b1804b1-442d6d6b6b3mr87886265e9.18.1746994414900;
        Sun, 11 May 2025 13:13:34 -0700 (PDT)
Received: from localhost.localdomain (93-34-88-225.ip49.fastwebnet.it.
 [93.34.88.225])
        by smtp.googlemail.com with ESMTPSA id
 5b1f17b1804b1-442d67ee275sm100615165e9.19.2025.05.11.13.13.33
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Sun, 11 May 2025 13:13:34 -0700 (PDT)
From: Christian Marangi <ansuelsmth@gmail.com>
To: Andrew Lunn <andrew+netdev@lunn.ch>,
	"David S. Miller" <davem@davemloft.net>,
	Eric Dumazet <edumazet@google.com>,
	Jakub Kicinski <kuba@kernel.org>,
	Paolo Abeni <pabeni@redhat.com>,
	Rob Herring <robh@kernel.org>,
	Krzysztof Kozlowski <krzk+dt@kernel.org>,
	Conor Dooley <conor+dt@kernel.org>,
	Lorenzo Bianconi <lorenzo@kernel.org>,
	Heiner Kallweit <hkallweit1@gmail.com>,
	Russell King <linux@armlinux.org.uk>,
	Philipp Zabel <p.zabel@pengutronix.de>,
	Nathan Chancellor <nathan@kernel.org>,
	Nick Desaulniers <nick.desaulniers+lkml@gmail.com>,
	Bill Wendling <morbo@google.com>,
	Justin Stitt <justinstitt@google.com>,
	Christian Marangi <ansuelsmth@gmail.com>,
	Daniel Golle <daniel@makrotopia.org>,
	netdev@vger.kernel.org,
	devicetree@vger.kernel.org,
	linux-kernel@vger.kernel.org,
	linux-arm-kernel@lists.infradead.org,
	linux-mediatek@lists.infradead.org,
	llvm@lists.linux.dev
Subject: [net-next PATCH v4 09/11] dt-bindings: net: pcs: Document support for
 Airoha Ethernet PCS
Date: Sun, 11 May 2025 22:12:35 +0200
Message-ID: <20250511201250.3789083-10-ansuelsmth@gmail.com>
X-Mailer: git-send-email 2.48.1
In-Reply-To: <20250511201250.3789083-1-ansuelsmth@gmail.com>
References: <20250511201250.3789083-1-ansuelsmth@gmail.com>
MIME-Version: 1.0
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20250511_131336_794050_64D53C7A 
X-CRM114-Status: GOOD (  13.35  )
X-BeenThere: linux-mediatek@lists.infradead.org
X-Mailman-Version: 2.1.34
Precedence: list
List-Id: <linux-mediatek.lists.infradead.org>
List-Unsubscribe: <http://lists.infradead.org/mailman/options/linux-mediatek>,
 <mailto:linux-mediatek-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-mediatek/>
List-Post: <mailto:linux-mediatek@lists.infradead.org>
List-Help: <mailto:linux-mediatek-request@lists.infradead.org?subject=help>
List-Subscribe: <http://lists.infradead.org/mailman/listinfo/linux-mediatek>,
 <mailto:linux-mediatek-request@lists.infradead.org?subject=subscribe>
Sender: "Linux-mediatek" <linux-mediatek-bounces@lists.infradead.org>
Errors-To: 
 linux-mediatek-bounces+linux-mediatek=archiver.kernel.org@lists.infradead.org

Document support for Airoha Ethernet PCS for AN7581 SoC.

Airoha AN7581 SoC expose multiple Physical Coding Sublayer (PCS) for
the various Serdes port supporting different Media Independent Interface
(10BASE-R, USXGMII, 2500BASE-X, 1000BASE-X, SGMII).

This follow the new PCS provider with the use of #pcs-cells property.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
---
 .../bindings/net/pcs/airoha,pcs.yaml          | 112 ++++++++++++++++++
 1 file changed, 112 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/net/pcs/airoha,pcs.yaml

--- /dev/null
+++ b/Documentation/devicetree/bindings/net/pcs/airoha,pcs.yaml
@@ -0,0 +1,112 @@
+# SPDX-License-Identifier: GPL-2.0-only OR BSD-2-Clause
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/net/pcs/airoha,pcs.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Airoha Ethernet PCS and Serdes
+
+maintainers:
+  - Christian Marangi <ansuelsmth@gmail.com>
+
+description:
+  Airoha AN7581 SoC expose multiple Physical Coding Sublayer (PCS) for
+  the various Serdes port supporting different Media Independent Interface
+  (10BASE-R, USXGMII, 2500BASE-X, 1000BASE-X, SGMII).
+
+properties:
+  compatible:
+    enum:
+      - airoha,an7581-pcs-eth
+      - airoha,an7581-pcs-pon
+
+  reg:
+    items:
+      - description: XFI MAC reg
+      - description: HSGMII AN reg
+      - description: HSGMII PCS reg
+      - description: MULTI SGMII reg
+      - description: USXGMII reg
+      - description: HSGMII rate adaption reg
+      - description: XFI Analog register
+      - description: XFI PMA (Physical Medium Attachment) register
+
+  reg-names:
+    items:
+      - const: xfi_mac
+      - const: hsgmii_an
+      - const: hsgmii_pcs
+      - const: multi_sgmii
+      - const: usxgmii
+      - const: hsgmii_rate_adp
+      - const: xfi_ana
+      - const: xfi_pma
+
+  resets:
+    items:
+      - description: MAC reset
+      - description: PHY reset
+
+  reset-names:
+    items:
+      - const: mac
+      - const: phy
+
+  "#pcs-cells":
+    const: 0
+
+required:
+  - compatible
+  - reg
+  - reg-names
+  - resets
+  - reset-names
+  - "#pcs-cells"
+
+additionalProperties: false
+
+examples:
+  - |
+    #include <dt-bindings/reset/airoha,en7581-reset.h>
+
+    pcs@1fa08000 {
+        compatible = "airoha,an7581-pcs-pon";
+        reg = <0x1fa08000 0x1000>,
+              <0x1fa80000 0x60>,
+              <0x1fa80a00 0x164>,
+              <0x1fa84000 0x450>,
+              <0x1fa85900 0x338>,
+              <0x1fa86000 0x300>,
+              <0x1fa8a000 0x1000>,
+              <0x1fa8b000 0x1000>;
+        reg-names = "xfi_mac", "hsgmii_an", "hsgmii_pcs",
+                    "multi_sgmii", "usxgmii",
+                    "hsgmii_rate_adp", "xfi_ana", "xfi_pma";
+
+        resets = <&scuclk EN7581_XPON_MAC_RST>,
+                 <&scuclk EN7581_XPON_PHY_RST>;
+        reset-names = "mac", "phy";
+
+        #pcs-cells = <0>;
+    };
+
+    pcs@1fa09000 {
+        compatible = "airoha,an7581-pcs-eth";
+        reg = <0x1fa09000 0x1000>,
+              <0x1fa70000 0x60>,
+              <0x1fa70a00 0x164>,
+              <0x1fa74000 0x450>,
+              <0x1fa75900 0x338>,
+              <0x1fa76000 0x300>,
+              <0x1fa7a000 0x1000>,
+              <0x1fa7b000 0x1000>;
+        reg-names = "xfi_mac", "hsgmii_an", "hsgmii_pcs",
+                    "multi_sgmii", "usxgmii",
+                    "hsgmii_rate_adp", "xfi_ana", "xfi_pma";
+
+        resets = <&scuclk EN7581_XSI_MAC_RST>,
+                 <&scuclk EN7581_XSI_PHY_RST>;
+        reset-names = "mac", "phy";
+
+        #pcs-cells = <0>;
+    };
